api/openapi-spec/swagger.json

```diff
@@ -5003,7 +5003,12 @@
           "type": "string"
         },
         "protocol": {
-          "description": "Protocol for port. Must be UDP, TCP, or SCTP. Defaults to \"TCP\".",
+          "description": "Protocol for port. Must be UDP, TCP, or SCTP. Defaults to \"TCP\".\n\nPossible enum values:\n - `SCTP`: is the SCTP protocol.\n - `TCP`: is the TCP protocol.\n - `UDP`: is the UDP protocol.",
```

Comment:
Super cool that this comes for free.  It does suggest we need better norms for our API comments - this reads really  awkwardly IMO.  As an example it's not great - the comments on the values are totally pointless and should, IMO, be removed.  Go-style comments "<symbol> is ..." make this awkward too.

I'd love to see 2 examples:
1) Where the values have no comments (as it should be for Protocol).
2) Where the values have long comments.

Then we could  define conventions for comments.  We have competing requirements - we want it to render well in godoc but also in openapi...

Reply:
> Super cool that this comes for free. It does suggest we need better norms for our API comments - this reads really awkwardly IMO. As an example it's not great - the comments on the values are totally pointless and should, IMO, be removed. Go-style comments " is ..." make this awkward too.

Totally agree with that,

> Then we could define conventions for comments. We have competing requirements - we want it to render well in godoc but also in openapi...

I feel like both should be able to deal with splitting long lines on their end rather than trying to do a "one-size-fits-all", especially openapi which isn't meant to be consumed directly. What did you have in mind specifically when you said that?

That being said, I can see how:
1. We should probably not warp lines that exceeds 72 chars in the openapi (but since go code/comments tends to be warped, we may need to unwarp, the same way godoc does).
2. We probably should preserve empty lines that forms separate block.
3. We need to be mindful of lists/enumerations (we shouldn't unwarp these).

All-in-all, there are ways to be smart here, good points, thanks @thockin!